src/textual/screen.py

```diff
@@ -939,6 +939,25 @@ def action_dismiss(
         """
         self.dismiss(result)
 
+    def can_view(self, widget: Widget) -> bool:
+        """Check if a given widget is in the current view (scrollable area).
+
+        Note: This doesn't necessarily equate to a widget being visible.
+        There are other reasons why a widget may not be visible.
```

Comment:
I remember being bitten by this before.
Is it worth giving an example of a reason why?
Otherwise, I'm just left wondering...

Reply:
That might be a good PR for someone to do to `Widget.can_view`, as this is just overriding it for a slightly unrelated thing.

Reply:
Right. But I'd still like for you to give me an example! 😆  If you don't mind. I can't think of something now.

Reply:
I'm not personally aware of one, especially not being the author of the method this is overriding. Here I'm just carrying through the documentation.

Reply:
Not sure if I'm following this chat correctly but:

- If the widget's on a different `layer` and has an offset which means it no longer falls within the region of its parent. The offset may push it off-screen for example.
- If the widget is has "hidden" `visibility`. It's still within the scrollable area, but it's not visible.